jumpscripts/maintenance/fix_rebooted_nodes.py

```diff
@@ -0,0 +1,80 @@
+from JumpScale import j
+import time
+
+descr = """
+This script ensures all vms and router oses survive a node reboot.
+"""
+
+organization = "jumpscale"
+author = "tareka@greenitglobe.com"
+category = "monitor.maintenance"
+license = "bsd"
+version = "1.0"
+period = 180  # always in sec
+timeout = period * 0.2
+startatboot = True
+order = 1
+enable = True
+queue = 'process'
+roles = ['cpunode']
+
+
+def action():
+    exit_code, out = j.system.process.execute('cat /proc/uptime')
+    if exit_code != 0:
+        raise RuntimeError('Error getting uptime from cpunode')
+    up_time = out.strip('\n').split(' ')
+    if int(up_time[0]) > 300:
+        return
+
+    #get model data ( virtual machines and router oses)
+    import libvirt
+    nid = j.application.whoAmI.nid
+    scl = j.clients.osis.getNamespace('system')
+    ccl = j.clients.osis.getNamespace('cloudbroker')
+    vcl = j.clients.osis.getNamespace('vfw')
+    pcl = j.clients.portal.getByInstance('main')
+    result = ccl.stack.searchOne({'referenceId': str(nid)})
+    if result:
+        stack_id = result['id']
+    else:
+        raise RuntimeError('This node id does not belong to a cpu node please check where you are running this script')
+    # get all running virtual machines on this node
+    model_virtual_machines = ccl.vmachine.search({'status': 'RUNNING', 'stackId': stack_id})[1:]
+    for machine in ccl.vmachine.search({'status': 'HALTED',  'stackId': stack_id})[1:]:
+        results = scl.audit.search({"tags": {"$regex":'.*machineId:%s.*' % machine['id']},
```

Comment:
there shoudl a be a filter when this happened. like this it could have happend days ago...

Reply:
i get the epoch difference in the next lines isnt it going to be much more than the three hundred if it was days ago , or should i add that to the query

Reply:
@abdulrahmantkhalifa yes but in such a case maintenance mode should have kicked in